src/components/signUpComponent.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'proptypes';
+import Input from './input';
+import Loader from './loader';
+import logo from '../assets/img/logo.png';
+
+class SignUpComponent extends Component {
+    state = {
+        username: '',
+        email: '',
+        password: '',
+        confirm_password: '',
+    }
+
+    handleChange = (event) => {
+        this.setState({ [event.target.name]: event.target.value});
+    }
+
+    onSignUp = (event) => {
+        event.preventDefault();
+        const { password, confirm_password } = this.state;
+        if (password === confirm_password) {
+          const { signup } = this.props;
+          signup(this.state);
+        } else {
+          this.setState({ passwordError: 'Passwords don\'t match.' });
+        }
+      }
+
+    render () {
+        const { loading, success, error } = this.props;
+        const { errors } = error ? error.response.data : { errors: '' };
+        const { username, email, password, confirm_password } = this.state;
+        return(
+            <div id="body" className="signup-body">
+                <div className="form-content">
+                    {success ? <Redirect to="/login.html" /> : null}
+                    <div className="logo">
+                        <img src={ logo } alt="Blogger"/>
+                    </div>
+                    <form onSubmit={this.onSignUp}>
+                        <Input
+                            type="text"
+                            name="username"
+                            id="username"
+                            placeholder="Username"
+                            onChange={this.handleChange}
+                            defaultValue={username}
```

Comment:
No need to use the `defaultValue` with an onChange handler. Either use `value` with an onChange, or defaultValue with a ref. In the first instance, the component state will be controlled by React(us), in the second instance, the state is maintained by the component(DOM) (uncontrolled).
https://reactjs.org/docs/uncontrolled-components.html 
https://reactjs.org/docs/forms.html#controlled-components